compiler/dyno/lib/types/QualifiedType.cpp

```diff
@@ -50,6 +51,22 @@ bool QualifiedType::isParamFalse() const {
   return false;
 }
 
+bool QualifiedType::isParamKnownTuple() const {
+  if (kind_ == Kind::PARAM && type_ != nullptr) {
+    if (auto tt = type_->toTupleType()) {
+      // Check if all type actuals are known params
+      for (int i = 0; i < tt->numElements(); i++) {
```

Comment:
I think this is fine for now, but in the future we should compute this once in the TupleType constructor and cache the result.

Reply:
I figured it was easy enough to fix 😄